packages/table/src/quadrants/tableQuadrantStack.tsx

```diff
@@ -536,48 +533,59 @@ export class TableQuadrantStack extends AbstractComponent<ITableQuadrantStackPro
+
+        // update this flag before updating the main quadrant scroll offsets,
+        // since we need this set before onScroll fires.
+        if (nextScrollLeft != null || nextScrollTop != null) {
+            this.wasMainQuadrantScrollChangedFromOtherOnWheelCallback = true;
+        }
 
+        // note that these upcoming DOM writes are batched together after the reads above.
+        this.maybeSetGlobalScrollOffset("scrollLeft", nextScrollLeft);
+        this.maybeSetGlobalScrollOffset("scrollTop", nextScrollTop);
+
+        // sync less important view stuff when scrolling/wheeling stops.
         this.syncQuadrantViewsDebounced();
     };
 
-    private handleDirectionalWheel = (
-        direction: "horizontal" | "vertical",
-        delta: number,
-        quadrantTypesToSync: QuadrantType[],
-    ) => {
-        const isHorizontal = direction === "horizontal";
+    private getNextScrollOffset = (direction: "horizontal" | "vertical", delta: number) => {
+        const { grid, isHorizontalScrollDisabled, isVerticalScrollDisabled } = this.props;
 
+        const isHorizontal = direction === "horizontal";
         const scrollKey = isHorizontal ? "scrollLeft" : "scrollTop";
-        const isScrollDisabled = isHorizontal
-            ? this.props.isHorizontalScrollDisabled
-            : this.props.isVerticalScrollDisabled;
-
-        if (!isScrollDisabled) {
-            this.wasMainQuadrantScrollChangedFromOtherOnWheelCallback = true;
+        const isScrollDisabled = isHorizontal ? isHorizontalScrollDisabled : isVerticalScrollDisabled;
 
-            const mainScrollContainer = this.quadrantRefs[QuadrantType.MAIN].scrollContainer;
-            const currScrollOffset = mainScrollContainer[scrollKey];
-            const nextScrollOffset = Math.max(0, mainScrollContainer[scrollKey] + delta);
+        if (isScrollDisabled) {
+            return undefined;
+        }
 
-            if (nextScrollOffset === currScrollOffset) {
-                return;
-            }
+        // measure client size on the first event of the current wheel gesture,
+        // then grab cached values on successive events to eliminate DOM reads.
+        // requires clearing the cached values in the debounced view-update at
+        // the end of the wheel event.
+        // ASSUMPTION: the client size won't change during the wheel event.
+        let clientSize = isHorizontal
+            ? this.cache.getScrollContainerClientWidth()
+            : this.cache.getScrollContainerClientHeight();
+
+        if (clientSize == null) {
+            // should trigger only on the first scroll of the wheel gesture
+            clientSize = this.updateScrollContainerClientSize(isHorizontal);
+        }
 
-            mainScrollContainer[scrollKey] = nextScrollOffset;
+        const containerSize = isHorizontal
+            ? this.cache.getScrollContainerClientWidth() - this.cache.getRowHeaderWidth()
```

Comment:
this code is a little difficult to read because above you access `getScrollContainerClientWidth` which according the conditional on line 574, could be nully. then right after, you invoke them again without a null check. this implies that `updateScrollContainerClientSize` will be setting the right value, but in order to verify, i have to ready that method and look at it's conditionals.